mm2src/coins/nft/nft_tests.rs

```diff
@@ -1,194 +1,667 @@
-        let meta_redacted = "{\"name\":\"URL redacted for user protection\",\"image\":\"https://tikimetadata.s3.amazonaws.com/tiki_box.png\"}";
-        assert_eq!(meta_redacted, nft.common.metadata.unwrap())
-    }
-
-    #[test]
-    fn test_moralis_requests() {
-        let response_nft_list = block_on(send_request_to_uri(NFT_LIST_URL_TEST)).unwrap();
-        let nfts_list = response_nft_list["result"].as_array().unwrap();
-        for nft_json in nfts_list {
-            let nft_moralis: NftFromMoralis = serde_json::from_str(&nft_json.to_string()).unwrap();
-            assert_eq!(TEST_WALLET_ADDR_EVM, eth_addr_to_hex(&nft_moralis.common.owner_of));
-        }
-
-        let response_transfer_history = block_on(send_request_to_uri(NFT_HISTORY_URL_TEST)).unwrap();
-        let mut transfer_list = response_transfer_history["result"].as_array().unwrap().clone();
-        assert!(!transfer_list.is_empty());
-        let first_transfer = transfer_list.remove(transfer_list.len() - 1);
-        let transfer_moralis: NftTransferHistoryFromMoralis =
-            serde_json::from_str(&first_transfer.to_string()).unwrap();
-        assert_eq!(
-            TEST_WALLET_ADDR_EVM,
-            eth_addr_to_hex(&transfer_moralis.common.to_address)
-        );
-
-        let response_meta = block_on(send_request_to_uri(NFT_METADATA_URL_TEST)).unwrap();
-        let nft_moralis: NftFromMoralis = serde_json::from_str(&response_meta.to_string()).unwrap();
-        assert_eq!(41237364, *nft_moralis.block_number_minted.unwrap());
-        let token_uri = nft_moralis.common.token_uri.unwrap();
-        let uri_response = block_on(send_request_to_uri(token_uri.as_str())).unwrap();
-        serde_json::from_str::<UriMeta>(&uri_response.to_string()).unwrap();
-    }
-
-    #[test]
-    fn test_add_get_nfts() { block_on(test_add_get_nfts_impl()) }
-
-    #[test]
-    fn test_last_nft_blocks() { block_on(test_last_nft_blocks_impl()) }
+#[cfg(not(target_arch = "wasm32"))]
+use mm2_net::native_http::send_request_to_uri;
 
-    #[test]
-    fn test_nft_list() { block_on(test_nft_list_impl()) }
-
-    #[test]
-    fn test_remove_nft() { block_on(test_remove_nft_impl()) }
-
-    #[test]
-    fn test_refresh_metadata() { block_on(test_refresh_metadata_impl()) }
-
-    #[test]
-    fn test_nft_amount() { block_on(test_nft_amount_impl()) }
-
-    #[test]
-    fn test_add_get_transfers() { block_on(test_add_get_transfers_impl()) }
-
-    #[test]
-    fn test_last_transfer_block() { block_on(test_last_transfer_block_impl()) }
-
-    #[test]
-    fn test_transfer_history() { block_on(test_transfer_history_impl()) }
+common::cfg_wasm32! {
+    use wasm_bindgen_test::*;
+    wasm_bindgen_test::wasm_bindgen_test_configure!(run_in_browser);
+    use mm2_net::wasm_http::send_request_to_uri;
+}
 
-    #[test]
-    fn test_transfer_history_filters() { block_on(test_transfer_history_filters_impl()) }
+macro_rules! cross_test {
```

Comment:
You can move this macro to common module instead of duplicating it https://github.com/KomodoPlatform/komodo-defi-framework/blob/a2cb99bfff328da74f9fb5cdfc877ffa1ef7bafe/mm2src/mm2_event_stream/src/controller.rs#L112-L122

Reply:
Done